File: Move_Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        I = 0
        J = 1
        while J < len(nums):
            if nums[I] == 0:
                if nums[J] != 0:
                    nums[I], nums[J] = nums[J], nums[I]
                    I += 1
                J += 1
            else:
                I += 1
                if J <= I:
                    J = I + 1

File: Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int lastNonZeroFoundAt = 0;
        int i = 0;
        int n = nums.size();
        int temp;

        while(i<n){
            if(nums[i]!=0){
                temp = nums[lastNonZeroFoundAt];
                nums[lastNonZeroFoundAt] = nums[i];
                nums[i] = temp;
                lastNonZeroFoundAt += 1;
            }
            i += 1;
        }
    }
};
